backend/src/middleware/auth.ts

```
@@ -0,0 +1,345 @@
+import { prisma } from '@/config/database';
+import { AuthenticatedRequest, AuthUser, JwtPayload, UserRole } from '@/types';
+import logger, { loggers } from '@/config/logger';
+
+// Extend Request interface to include user
+declare global {
+  namespace Express {
+    interface Request {
+      user?: any;
+      userId?: string;
+    }
+  }
```

ESLint

[error] ES2015 module syntax is preferred over namespaces.

(@typescript-eslint/no-namespace)